### PR TITLE
add bernoulli op

### PR DESCRIPTION
**Pull request by zhiqiu**
### PR types

New features
### PR changes

APIs
### Describe

- add `bernoulli` op （c++ OP）
- add `paddle.bernoulli(x)`  (python API)

### CODE REVIEW

File: paddle/fluid/operators/bernoulli_op.cu [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+namespace operators {
+// it can be consistent with cpu when CUDAGenerator is provided.
+template <typename T>
+struct BernoulliCudaFunctor {
+  unsigned int seed_;
+  __host__ __device__ BernoulliCudaFunctor(int seed) : seed_(seed) {}
+
+  __host__ __device__ T operator()(const unsigned int n, const T p) const {
+    thrust::minstd_rand rng;
+    rng.seed(seed_);
```
Comment (jzhang533) [inline paddle/fluid/operators/bernoulli_op.cu:35]: will change to use global generator later

**Review by jzhang533 — DISMISSED** [review]
lgtm

File: paddle/fluid/operators/bernoulli_op.cu (hunk 1 above)
Comment (zhiqiu) [reply paddle/fluid/operators/bernoulli_op.cu:35]: Yes, it will be refined after GPUGenerator is provided.

File: python/paddle/tensor/random.py [hunk 2]
```diff
@@ -37,6 +38,51 @@
 ]
 
 
+def bernoulli(x):
```
Comment (lanxianghit) [inline python/paddle/tensor/random.py]: 这个API是否需要name参数

File: python/paddle/tensor/random.py [hunk 3]
```diff
@@ -37,6 +38,51 @@
...
+    This OP returns a Tensor filled with random binary(0 or 1) number from a Bernoulli distribution.
+    The input ``x`` is a tensor with probabilities for generating the random binary number.
+    Each element in ``x`` should be in [0, 1], and the out is generated by:
+    
+    .. math::
+
+        out_i ~ Bernoulli (x_i)
+
+    Args:
+        x(int):  A tensor with probabilities for generating the random binary number. The data type
```
Comment (lanxianghit) [inline python/paddle/tensor/random.py]: x(int) ?

File: python/paddle/tensor/random.py [hunk 4]
```diff
@@ -37,6 +38,51 @@
...
+
+        import paddle
+        import numpy as np
+
+        paddle.disable_static()
+
+        x = paddle.rand([2, 3])
+        print(x.numpy())
+        out = paddle.bernoulli(x)
+        print(out.numpy())
```
Comment (lanxianghit) [inline python/paddle/tensor/random.py:75]: 给一下输出结果

File: python/paddle/tensor/random.py [hunk 5]
```diff
@@ -37,6 +38,51 @@
...
+    """
+
+    if in_dygraph_mode():
+        return core.ops.bernoulli(x)
+
+    check_variable_and_dtype(x, "x", ["float32", "float64"], "bernoulli")
+
+    helper = LayerHelper("randint", **locals())
+    out = helper.create_variable_for_type_inference(
+        dtype=x.dtype)  # maybe set out to int32 ?
```
Comment (lanxianghit) [inline python/paddle/tensor/random.py:88]: # maybe set out to int32 ? 

是否需要有结论？

**Comment by zhiqiu** [comment]
Thanks, @lanxianghit  I will refine the doc in next PR.

**Review by lanxianghit — DISMISSED** [review]

File: python/paddle/tensor/random.py (hunk 2 above)
Comment (zhiqiu) [reply python/paddle/tensor/random.py]: Done

File: python/paddle/tensor/random.py (hunk 3 above)
Comment (zhiqiu) [reply python/paddle/tensor/random.py]: Thanks, changed to x(Tensor).

File: python/paddle/tensor/random.py (hunk 4 above)
Comment (zhiqiu) [reply python/paddle/tensor/random.py:75]: Done

**Review by lanxianghit — APPROVED** [review]

**Review by jzhang533 — APPROVED** [review]
lgtm

**Pull request merged**